File: springboot/security/oauth2_basic_board/src/main/java/com/example/oauth2_basic_board/domain/repository/BoardRepository.java
package com.example.oauth2_basic_board.domain.repository;

import com.example.basicboard_token.domain.entity.Board;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface BoardRepository extends JpaRepository<Board, Long>, BoardRepositoryCustom {
    @Query("""
        select b
        from Board b
        left join fetch b.comments
        where b.id = :id
    """)
    Optional<Board> findDetailById(Long id);
}

File: kotlin/kotlin-part-4/src/main/kotlin/org/example/kotlinpart4/domain/repository/BoardRepository.kt
package org.example.kotlinpart4.domain.repository

import org.example.kotlinpart4.domain.entity.Board
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository

interface BoardRepository: JpaRepository<Board, Long> {
    // 쿼리 메서드
    // - 메서드 이름을 스프링 데이터가 해석해서 SQL을 만들어준다
    // containing : where title like %?%
    fun findByTitleContaining(keyword: String, pageable: Pageable): Page<Board>
}